### ISSUE TITLE
Feed Neural strategy more info from other indicators

### ISSUE THREAD

**Issue by glennfu** [DeviaVir/zenbot#1005]
Tagging @darthyvader and @JensvdHeydt since they seem to have expertise on the neural strategy.

It occurs to me that as a strategy competing with others like macd and crossover_vwap, the neural strategy as written is good. However, a smarter neural net strategy to really perform well should actually feed in data from other indicators, along with data points like day of week and time of day. Maybe even points like high/low of the day, high/low of the week, high/low of the month. Rather than expect it to be a replacement for statistical analysis, it should actually leverage results of statistical analysis for its own training.

Any thoughts on whether or not this is reasonable, or feasible?

**Comment by DeviaVir** [comment]
This issue was moved to highvelocityspace/zenbot#9

**Comment by slashafk** [comment]
Hello,
The neural strategy is only a regression AI so far, so I don't think in it's actual state it could work. 

A new approach should be used, in this case a classification AI with 3 classes : BUY, HOLD, SELL.
(http://cs.stanford.edu/people/karpathy/convnetjs/demo/mnist.html).

Convnetjs could be used for that with few adjustment to the actual neural strategy switching : 
- the activation function to softmax with num_classes to 3
- the input to an array of all our indicators (RSI, CCI, ... to reduce the time of learning I'd suggest to transform those in boolean : [current_price,low,high,close,open,RSI>60,RSI<40,CCI<100,CCI>100,...]

With such systems I hardly think that the convnetjs and 100 learning would be enough for the AI to use some EMA, MACD, ...learning the relation between the price and those values could be complicated.

Also the architecture of zenbot could be a problem, zenbot push all the actual status (including AI) to the lookback increasing the memory usage on each tick with a macd,cci,rsi, ... strategy it's ok, they're is not that much elements to store but the AI could use lots of memory.
